### PR TITLE
added proxy logs, added events logs

### PR DESCRIPTION
**Pull request by RoyUP9**

### CODE REVIEW

File: cli/cmd/common.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ func startProxyReportErrorIfAny(kubernetesProvider *kubernetes.Provider, cancel
 			"Try setting different port by using --%s", errormessage.FormatError(err), configStructs.GuiPortTapName))
 		cancel()
 	}
+
+	logger.Log.Debugf("done starting proxy")
```
Comment (IgorGov) [inline cli/cmd/common.go:30]: it is not "done starting proxy" -> it is more like "proxy ended"

File: cli/cmd/tapRunner.go [hunk 2]
```diff
@@ -497,7 +498,8 @@ func watchApiServerPod(ctx context.Context, kubernetesProvider *kubernetes.Provi
 				isPodReady = true
 				go startProxyReportErrorIfAny(kubernetesProvider, cancel)
 
-				if err := apiserver.Provider.InitAndTestConnection(GetApiServerUrl(), 20); err != nil {
+				retries := config.GetEnvConfig(reflect.Int, config.ApiServerRetries, reflect.ValueOf(20))
```
Comment (IgorGov) [inline cli/cmd/tapRunner.go:501]: it is something that should be inside the provider as its internal configuration

File: cli/config/envConfig.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+func GetEnvConfig(kind reflect.Kind, key string, defaultValue reflect.Value) reflect.Value {
+	val := os.Getenv(key)
+	if val == "" {
+		return defaultValue
+	}
+
+	parsedValue, err := getParsedValue(kind, val)
+	if err != nil {
+		logger.Log.Warningf(uiUtils.Warning, fmt.Sprintf("error parsing env value, expected value: %v, key: %v, value: %v", kind, key, val))
```
Comment (IgorGov) [inline cli/config/envConfig.go:23]:
```suggestion
		logger.Log.Warningf(uiUtils.Warning, fmt.Sprintf("error parsing env value, expected value type: %v, key: %v, value: %v", kind, key, val))
```

File: cli/mizu/fsUtils/mizuLogsUtils.go [hunk 4]
```diff
@@ -39,22 +39,39 @@ func DumpLogs(provider *kubernetes.Provider, ctx context.Context, filePath strin
...
 		if err := AddStrToZip(zipWriter, logs, fmt.Sprintf("%s.%s.log", pod.Namespace, pod.Name)); err != nil {
 			logger.Log.Errorf("Failed write logs, %v", err)
 		} else {
 			logger.Log.Debugf("Successfully added log length %d from pod: %s.%s", len(logs), pod.Namespace, pod.Name)
 		}
 	}
+
+	events, err := provider.GetNamespaceEvents(config.Config.MizuResourcesNamespace, ctx)
+	if err != nil {
+		logger.Log.Errorf("Failed to get events, %v", err)
```
Comment (IgorGov) [inline cli/mizu/fsUtils/mizuLogsUtils.go:52]: debug not error

File: cli/mizu/fsUtils/mizuLogsUtils.go [hunk 5]
```diff
@@ -39,22 +39,39 @@ func DumpLogs(provider *kubernetes.Provider, ctx context.Context, filePath strin
...
+	} else {
+		logger.Log.Debugf("Successfully read events for namespace: %s", config.Config.MizuResourcesNamespace)
+	}
+
+	if err := AddStrToZip(zipWriter, events, fmt.Sprintf("%s_events.log", config.Config.MizuResourcesNamespace)); err != nil {
+		logger.Log.Errorf("Failed write logs, %v", err)
```
Comment (IgorGov) [inline cli/mizu/fsUtils/mizuLogsUtils.go:58]:
```suggestion
		logger.Log.Debugf("Failed write logs, %v", err)
```

File: cli/cmd/common.go (hunk 1 above)
Comment (RoyUP9) [reply cli/cmd/common.go:30]: but the proxy is not shutting down, it started running at that point

File: cli/cmd/tapRunner.go (hunk 2 above)
Comment (RoyUP9) [reply cli/cmd/tapRunner.go:501]: i am not sure i understood what you meant

File: cli/kubernetes/provider.go [hunk 6]
```diff
@@ -739,6 +739,16 @@ func (provider *Provider) GetPodLogs(namespace string, podName string, ctx conte
 	return str, nil
 }
 
+func (provider *Provider) GetNamespaceEvents(namespace string, ctx context.Context) (string, error) {
+	eventsOpts := metav1.ListOptions{}
+	eventList, err := provider.clientSet.CoreV1().Events(namespace).List(ctx, eventsOpts)
+	if err != nil {
+		return "", fmt.Errorf("error getting events on ns: %s, %w", namespace, err)
```
Comment (IgorGov) [inline cli/kubernetes/provider.go:746]: should be debug

Comment (RoyUP9) [reply cli/kubernetes/provider.go:746]: but it's not a log

**Review by IgorGov — APPROVED** [review]

**Pull request merged**